pkg/querier/queryrange/roundtrip.go

```diff
@@ -63,6 +66,17 @@ func NewTripperware(cfg Config, log log.Logger, limits Limits, registerer promet
 				// weavework server uses httpgrpc errors for status code.
 				return nil, httpgrpc.Errorf(http.StatusBadRequest, err.Error())
 			}
+
+			userID, err := user.ExtractOrgID(req.Context())
```

Comment:
You should move this into the `if logSelector, ok := expr.(logql.LogSelectorExpr); ok {` we want to apply limits to only to log queries.  Also I suggest creating a new function to avoid a huge function like we have.

Reply:
We might want to have a test too.